simtools/applications/generate_corsika_histograms.py

```diff
@@ -0,0 +1,425 @@
+    """
+    Auxiliary function to run the histograms for the arguments given by event_1D_histograms.
+
+    Parameters
+    ----------
+    instance: `CorsikaOutput` instance.
+        The CorsikaOutput instance created in main.
+    output_dir: str
+        The output directory where to save the histograms.
+    event_1D_header_keys: str
+        Produce 1D histograms for elements given in `event_1D_header_keys` from the CORSIKA event
+        header and save into ecsv/png files.
+    png: bool
+        If true, histograms are saved into png files.
+    ecsv: bool
+        If true, histograms are saved into ecsv files.
+    """
+    for event_header_element in event_1D_header_keys:
+        if png:
+            figure, figure_name = corsika_output_visualize.plot_1D_event_header_distribution(
+                instance, event_header_element
+            )
+            output_file_name = Path(output_dir).joinpath(figure_name)
+            logger.info(f"Saving histogram to {output_file_name}")
+            figure.savefig(output_file_name, bbox_inches="tight")
+        if ecsv:
+            instance.export_event_header_1D_histogram(
+                event_header_element, bins=50, hist_range=None
+            )
+
+
+def _run_event_2D_histograms(instance, output_dir, event_2D_header_keys, png, ecsv):
+    """
+    Auxiliary function to run the histograms for the arguments given by event_1D_histograms.
+
+    Parameters
+    ----------
+    instance: `CorsikaOutput` instance.
+        The CorsikaOutput instance created in main.
+    output_dir: str
+        The output directory where to save the histograms.
+    event_2D_header_keys: str
+        Produce 1D histograms for elements given in `event_1D_header_keys` from the CORSIKA event
+        header and save into ecsv/png files.
+    png: bool
+        If true, histograms are saved into png files.
+    ecsv: bool
+        If true, histograms are saved into ecsv files.
+    """
+    for i_event_header_element, _ in enumerate(event_2D_header_keys[::2]):
+        if png:
+            figure, figure_name = corsika_output_visualize.plot_2D_event_header_distribution(
+                instance,
+                event_2D_header_keys[i_event_header_element],
+                event_2D_header_keys[i_event_header_element + 1],
+            )
+            output_file_name = Path(output_dir).joinpath(figure_name)
+            logger.info(f"Saving histogram to {output_file_name}")
+            figure.savefig(output_file_name, bbox_inches="tight")
+        if ecsv:
+            instance.export_event_header_2D_histogram(
+                event_2D_header_keys[i_event_header_element],
+                event_2D_header_keys[i_event_header_element + 1],
+                bins=50,
+                hist_range=None,
+            )
+
+
+def main():
+
+    label = Path(__file__).stem
+    description = "Generate histograms for the Cherenkov photons saved in the CORSIKA IACT file."
+    usage = ""
```

Comment:
I am not sure. We usually don't write usage. I believe in order not to duplicate the usage already given in the docstring

Reply:
So I actually suggest to remove it so that it defaults to `None`. The python `argparse` package would then know to build the correct usage command. When you provide an empty one it would be printed instead of the the correct one (check for yourself).